### DISCUSSION TITLE
Grafana 8.3.6 MySQL broken after update - all panels show "no data"

### DISCUSSION THREAD

**Discussion by lakelounge** [grafana/grafana#45467] [Data Sources]
Hi, i updated Grafana to 8.3.6 today (Raspberry Pi, Buster, all updates). Since this update all my panels show "no data". Checked the data sources (MySQL) but if i try to "save & test" one, i get „Load failed“. Anyone else experiencing this behavior? Thanks!

**Comment by lakelounge** [comment]
This is part of an endless logging in syslog … Maybe someone can point me in a direction …

Feb 16 19:02:26 raspberry3 grafana-server[8680]: sp      0x6e6ba8c
Feb 16 19:02:26 raspberry3 grafana-server[8680]: lr      0x2198280
Feb 16 19:02:26 raspberry3 grafana-server[8680]: pc      0x295994e
Feb 16 19:02:26 raspberry3 grafana-server[8680]: cpsr    0x60000010
Feb 16 19:02:26 raspberry3 grafana-server[8680]: fault   0x0
Feb 16 19:02:26 raspberry3 systemd[1]: grafana-server.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Feb 16 19:02:26 raspberry3 systemd[1]: grafana-server.service: Failed with result 'exit-code'.
Feb 16 19:02:26 raspberry3 systemd[1]: grafana-server.service: Service RestartSec=100ms expired, scheduling restart.
Feb 16 19:02:26 raspberry3 systemd[1]: grafana-server.service: Scheduled restart job, restart counter is at 8.
Feb 16 19:02:26 raspberry3 systemd[1]: Stopped Grafana instance.
Feb 16 19:02:26 raspberry3 systemd[1]: Started Grafana instance.
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=warn msg="falling back to legacy setting of 'min_interval_seconds'; please use the configuration option in the `unified_alerting` section if Grafana 8 alerts are enabled." logger=settings
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Config loaded from" logger=settings file=/usr/share/grafana/conf/defaults.ini
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Config loaded from" logger=settings file=/etc/grafana/grafana.ini
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.data=/var/lib/grafana"
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.logs=/var/log/grafana"
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.plugins=/var/lib/grafana/plugins"
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Config overridden from command line" logger=settings arg="default.paths.provisioning=/etc/grafana/provisioning"
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Path Home" logger=settings path=/usr/share/grafana
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Path Data" logger=settings path=/var/lib/grafana
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Path Logs" logger=settings path=/var/log/grafana
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Path Plugins" logger=settings path=/var/lib/grafana/plugins
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Path Provisioning" logger=settings path=/etc/grafana/provisioning
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="App mode production" logger=settings
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Connecting to DB" logger=sqlstore dbtype=sqlite3
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Starting DB migrations" logger=migrator
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="migrations completed" logger=migrator performed=0 skipped=438 duration=6.871987ms
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Validated license token" logger=licensing appURL=http://localhost:3000/ source=disk status=NotFound
Feb 16 19:02:26 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:26+0100 lvl=info msg="Initialising plugins" logger=plugin.manager
Feb 16 19:02:27 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:27+0100 lvl=info msg="Plugin registered" logger=plugin.manager pluginId=input
Feb 16 19:02:27 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:27+0100 lvl=info msg="Live Push Gateway initialization" logger=live.push_http
Feb 16 19:02:27 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:27+0100 lvl=info msg="Writing PID file" logger=server path=/run/grafana/grafana-server.pid pid=8718
Feb 16 19:02:27 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:27+0100 lvl=warn msg="Scheduling and sending of reports disabled, SMTP is not configured and enabled. Configure SMTP to enable." logger=report
Feb 16 19:02:27 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:27+0100 lvl=info msg="HTTP Server Listen" logger=http.server address=[::]:3000 protocol=http subUrl= socket=
Feb 16 19:02:30 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:30+0100 lvl=info msg="Request Completed" logger=context userId=2 orgId=1 uname=mc method=GET path=/api/live/ws status=0 remote_addr=192.168.178.20 time_ms=17 size=0 referer=
Feb 16 19:02:30 raspberry3 grafana-server[8718]: t=2022-02-16T19:02:30+0100 lvl=info msg="Initialized channel handler" logger=live channel=grafana/dashboard/uid/6dCD5gkgz address=grafana/dashboard/uid/6dCD5gkgz
Feb 16 19:03:25 raspberry3 kernel: [30184.225031] Unhandled prefetch abort: unknown 1 (0x001) at 0x0295994e
Feb 16 19:03:25 raspberry3 grafana-server[8718]: SIGBUS: bus error
Feb 16 19:03:25 raspberry3 grafana-server[8718]: PC=0x295994e m=7 sigcode=0
Feb 16 19:03:25 raspberry3 grafana-server[8718]: goroutine 467 [running]:
Feb 16 19:03:25 raspberry3 grafana-server[8718]: runtime: unknown pc 0x295994e
Feb 16 19:03:25 raspberry3 grafana-server[8718]: stack: frame={sp:0x70d3a8c, fp:0x0} stack=[0x70d2000,0x70d4000)
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3a0c:  0x00000000  0x07b72720  0x0007797c <runtime.stringtoslicebyte+0x0000009c>  0x07b72720
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3a1c:  0x00322f04 <regexp.compile+0x0000030c>  0x07b72720  0x07b726e8  0x00000001
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3a2c:  0x00000008  0x00322f5c <regexp.compile+0x00000364>  0x00000008  0x07b72720
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3a3c:  0x0208fe60 <github.com/grafana/grafana/pkg/tsdb/mysql.(*mySQLMacroEngine).Interpolate+0x0000006c>  0x07b726e8  0x00000001  0x00000024
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3a4c:  0x00000001  0x00000001  0x0477eec0  0x00000000
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3a5c:  0x00000000  0x00000000  0x07b72720  0x02198280
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3a6c:  0x070ae0c0  0x00000080  0x00000000  0x021980f4
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3a7c:  0x07918168  0x00000000  0x011c6044 <github.com/grafana/grafana/pkg/tsdb/sqleng.(*DataSourceHandler).executeQuery+0x000001dc>  0x0295994e
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3a8c: <0x0000001d  0x000000d4  0x07278a20  0x00000000
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3a9c:  0x00000000  0x00000000  0x00000000  0x00000000
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3aac:  0x0208ffe4 <github.com/grafana/grafana/pkg/tsdb/mysql.(*mySQLMacroEngine).Interpolate.func1+0x00000000>  0x070ae0c0  0x00000000  0x00000000
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3abc:  0x00000000  0x00000000  0x00000000  0x00000000
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3acc:  0x00000000  0x00000000  0x00000000  0x00000000
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3adc:  0x00000000  0x00000000  0x0008f994 <runtime.goexit+0x00000004>  0x070ae0c0
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3aec:  0x072787e0  0x2d0062c0  0x00000000  0xd99de1ea
Feb 16 19:03:25 raspberry3 grafana-server[8718]: 0x070d3afc:  0x0000000e  0x00000000  0x2d0062c0  0x00000000

**Comment by lakelounge** [comment]
i found out that this seems to happen only on MySQL / MariaDB Data Sources … InfluxDB Sources  seem to work.

**Comment by lakelounge** [comment]
After updating to Grafana 8.4 and rpi-eeprom 13.9-1~buster (both came out last night) my panels are up and running again.

**Comment by gabor** [comment]
i'm glad this worked out well at the end, closing the discussion.
